### PR TITLE
Start of arrow making script. Tested with boar-tusk arrowheads.

### PR DESCRIPTION
**Pull request by Dartellum**
Will create the arrowheads if the raw material is available. 

Will add check for arrowheads as well. I wanted to get this up here so people can test and provide feedback. 

Not ready for merging!

### CODE REVIEW

File: shape.lic [hunk 1]
```diff
@@ -129,11 +129,18 @@ class Shape
...
     echo('*** You will need to upgrade to a journeyman or master book before 176 ranks! ***') if DRSkill.getrank('Engineering') == 175
     turn_to("page #{find_recipe(@chapter, @recipe_name)}")
-    bput('study my book', 'Roundtime')
+    bput('study my book', 'Roundtime', 'Arrow shaft')
     stow_item('book')
     get_item('drawknife')
-    get_item("#{@material} lumber")
-    shape('scrape my lumber with my drawknife')
+    if @material == 'shafts'
+      get_item("#{@material}")
```
Comment (rpherbig) [inline shape.lic]: You don't need string interpolation here

**Comment by rpherbig** [comment]
✔️ Code looks good to me

**Comment by rpherbig** [comment]
> Not ready for merging!

Let me know when you think it's ready 😁

**Comment by Dartellum** [comment]
> > Not ready for merging!
> 
> Let me know when you think it's ready

Still need to check all the names and changes thereof for arrows from raw material. :)

File: shape.lic (hunk 1 above)
Comment (Dartellum) [reply shape.lic]: get_item("#{@material}") ==> get_item(@material) ?

File: arrows.lic [hunk 2]
```diff
@@ -0,0 +1,88 @@
+custom_require.call(%w[common common-crafting])
```
Comment (rpherbig) [inline arrows.lic]: 💡 Documentation header block

File: arrows.lic [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        { name: 'type', options: %w[couger-claw boar-tusk sabertooth angiswaerd hele'la basilisk elsralael soot-stained ice-adder jagged-horn drake-fang], variable: true, descriptions: "Which arrow type to create?" },
+        { name: 'lumber', regex: /\w+/i, variable: true, description: 'Type of lumber used.' },
+        { name: 'tools', options: %w[true false], variable: true, description: 'Get tools from clerk?'}
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    @type = args.type
+    @lumber = args.lumber || 'maple'
```
Comment (rpherbig) [inline arrows.lic]: ❓ This arg isn't optional, can it ever be `nil`?

File: arrows.lic [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  end
+
+  def store_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'store'])
+  end
+
+  def order_lumber
+    return unless bput('tap my lumber', 'You tap', 'What were') == 'What were'
+    wait_for_script_to_complete('go2', ['8864'])
+    fput('order 10')
```
Comment (rpherbig) [inline arrows.lic]: 💡 Can any of these `fput`s be `bput`s?

File: arrows.lic (hunk 3 above)
Comment (Dartellum) [reply arrows.lic]: Good question. I will test more.

File: arrows.lic (hunk 4 above)
Comment (Dartellum) [reply arrows.lic]: Sure, I can change to bputs.

File: arrows.lic (hunk 3 above)
Comment (Dartellum) [reply arrows.lic:24]: Removed or logic.

File: arrows.lic (hunk 4 above)
Comment (Dartellum) [reply arrows.lic:47]: Changed to use bputs and common-crafting methods.

File: arrows.lic (hunk 2 above)
Comment (Dartellum) [reply arrows.lic:1]: Added header. Will added documentation once finished.

File: shape.lic (hunk 1 above)
Comment (Dartellum) [reply shape.lic:136]: Changed to get_item(@material).

File: arrows.lic [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  def get_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'get'])
+  end
+
+  def store_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'store'])
+  end
+
+  def order_lumber
+    return unless bput('tap my lumber', 'You tap', 'What were') == 'What were'
```
Comment (jandersson) [inline arrows.lic:45]: :bulb: `exists?('lumber')` ?

Comment (Dartellum) [reply arrows.lic:45]: I can change to that.

File: arrows.lic (hunk 4 above)
Comment (jandersson) [reply arrows.lic:47]: These are still fputs ?

File: arrows.lic (hunk 3 above)
Comment (jandersson) [reply arrows.lic:24]: There is still or logic in this version? Git push ?

File: arrows.lic [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    fput('order 10')
+    fput('order 10')
+    fput("get maple lumber from #{@bag}")
+    fput('combine my lumber')
+    fput('stow lumber')
+  end
+
+  def check_flights
+    return unless bput('tap my flights', 'You tap', 'What were') == 'What were'
+    fput('order 12')
```
Comment (jandersson) [inline arrows.lic:56]: Should this walk to the room where flights can be ordered?

File: arrows.lic [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    fput('order 12')
+    fput('stow flights')
+  end
+
+  def make_shafts
+    return unless bput('tap my shafts', 'You tap', 'What were') == 'What were'
```
Comment (jandersson) [inline arrows.lic:62]: :bulb: `exists?('shafts')`

File: arrows.lic (hunk 6 above)
Comment (Dartellum) [reply arrows.lic:56]: Yes, I think I relied on a previous step to get me there. I have not put the latest version up yet. I now have drake fangs to see if those have a name change to catch. Perhaps this weekend I will. :)

File: arrows.lic (hunk 7 above)
Comment (Dartellum) [reply arrows.lic:62]: Again, I can switch to that. :)

File: arrows.lic (hunk 5 above)
Comment (jandersson) [reply arrows.lic:45]: reminder to include DRCI and common-items for `exists?`

File: arrows.lic (hunk 6 above)
Comment (jandersson) [reply arrows.lic:56]: rgr :) just giving it a once over, noticed theres no walk_to at this step

**Comment by jandersson** [comment]
Good stuff man, i know a fletching script has been asked for

File: arrows.lic (hunk 6 above)
Comment (Dartellum) [reply arrows.lic:56]: Going to add it hometown support.

Comment (Dartellum) [reply arrows.lic:56]: Changed to order_item(@hometown_data['stock-room'], 10). I might even remove the 10 and put an entry in base-crafting for the item - or use a regex to match like find_recipe does.

**Comment by Dartellum** [comment]
> > > Not ready for merging!
> > 
> > 
> > Let me know when you think it's ready
> 
> Still need to check all the names and changes thereof for arrows from raw material. :)

I have all the raw materials in the script now and tested a few of them.

**Comment by Dartellum** [comment]
Did not mean to close...

**Comment by Dartellum** [comment]
I believe it is ready for merging at this point.

File: arrows.lic [hunk 8]
```diff
@@ -42,47 +61,70 @@ class Arrows
...
+    bput("stow my shafts in my #{@bag}", "You put")
   end
 
   def make_arrowheads
-    new_type = @type
-    new_type == 'boar tusk' if @type == 'boar-tusk'
-    return unless bput('tap my arrowheads', 'You tap', 'I could not') == 'I could not'
-    case bput("get #{@type} from my #{@bag}", 'You get', 'I could not', 'What were')
+    return if exists?('arrowheads')
+    arrow_head = { "cougar-claw" => "curved claw", "boar-tusk" => "boar tusk", "sabertooth" => "ivory fang", "angiswaerd" => "angiswaerd tooth", "hele'la" => "needlesharp tooth", "basilisk" => "basilisk fang", "elsralael" => "elsralael tooth", "soot-stained" => "soot-streaked fang", "ice-adder" => "adder fang", "jagged-horn" => "jagged horn", "drake-fang" => "drake fang" }
```
Comment (rpherbig) [inline arrows.lic:94]: 💡 This would be good to move to a constant at the top of the file

File: arrows.lic [hunk 9]
```diff
@@ -42,47 +61,70 @@ class Arrows
...
+  def clean_up
+    bput("stow shafts", "you pick up", "What were", "Stow what?")
+  end
+
+  def find_piece(match_string)
+    wait_for_script_to_complete('go2', [@hometown_data['stock-room']])
+    echo(match_string)
+    piece = DRC.bput('order', "^\\d+\\)\\.\\s\\s(?:some|a|an|stack of|5 pieces of|10 pieces of|1 batch of straight)?\\s*#{match_string}")
+    piece =~ /^(\d+)/
+    echo(piece)
```
Comment (rpherbig) [inline arrows.lic:118]: 💡 Ideally this (and line 115) should be guarded by something like `... if @debug_mode` or such

Comment (Dartellum) [reply arrows.lic:118]: I thought I took those out.

Comment (Dartellum) [reply arrows.lic:118]: I'll remove those in the next iteration.

**Pull request merged**